controller/controller.go

```diff
@@ -1404,10 +1404,14 @@ func (ctrl *ProvisionController) provisionClaimOperation(ctx context.Context, cl
 	if nodeName, ok := getString(claim.Annotations, annSelectedNode, annAlphaSelectedNode); ok {
 		if ctrl.nodeLister != nil {
 			selectedNode, err = ctrl.nodeLister.Get(nodeName)
+			// if node does not exist, remove volume.kubernetes.io/selected-node annotation
+			if apierrs.IsNotFound(err) {
+				delete(claim.Annotations, annSelectedNode)
```

Comment:
With logs and this pr change: https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/compare/master...sunnylovestiramisu:sig-storage-lib-external-provisioner:fixProvisionerWithLogs


``k describe pvc task-pv-claim``

```
  Normal   ExternalProvisioning   30s (x3 over 30s)  persistentvolume-controller                                                                      waiting for a volume to be created, either by external provisioner "pd.csi.storage.gke.io" or manually created by system administrator
  Warning  ProvisioningFailed     30s                pd.csi.storage.gke.io_e2e-test-songsunny-minion-group-6dnk_a248f4cc-9002-4021-a4b2-36b16d19c0b2  node "non-exist-node" not found
  Normal   Provisioning           30s                pd.csi.storage.gke.io_e2e-test-songsunny-minion-group-6dnk_a248f4cc-9002-4021-a4b2-36b16d19c0b2  External provisioner is provisioning volume for claim "default/task-pv-claim"
  Normal   ProvisioningSucceeded  26s                pd.csi.storage.gke.io_e2e-test-songsunny-minion-group-6dnk_a248f4cc-9002-4021-a4b2-36b16d19c0b2  Successfully provisioned volume pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef
```

``k logs -n gce-pd-csi-driver csi-gce-pd-controller-77df58ff84-j8zhr csi-provisione``


```
I0320 18:46:18.972132       1 controller.go:1334] =================== Sunny Test ===================
I0320 18:46:18.972161       1 controller.go:1338] provision "default/task-pv-claim" class "in-tree-volume": started
I0320 18:46:18.972210       1 controller.go:1388] ================================= nodeLister error is node "non-exist-node" not found =================================
I0320 18:46:18.972223       1 controller.go:1392] ================================= is not found error =================================
I0320 18:46:18.972230       1 controller.go:1451] =================== err is node "non-exist-node" not found ===================
I0320 18:46:18.972865       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"task-pv-claim", UID:"b677fa97-f016-4b0f-870b-054f1fb1a6ef", APIVersion:"v1", ResourceVersion:"1828", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' node "non-exist-node" not found
I0320 18:46:18.981944       1 controller.go:1470] provision "default/task-pv-claim" class "in-tree-volume": volume rescheduled because: node "non-exist-node" not found
I0320 18:46:18.981965       1 controller.go:1471] =================== volume rescheduled because: node "non-exist-node" not found ===================
I0320 18:46:18.981983       1 controller.go:1071] Stop provisioning, removing PVC b677fa97-f016-4b0f-870b-054f1fb1a6ef from claims in progress


I0320 18:46:18.982019       1 controller.go:1334] =================== Sunny Test ===================
I0320 18:46:18.982034       1 controller.go:1338] provision "default/task-pv-claim" class "in-tree-volume": started
I0320 18:46:18.982060       1 controller.go:560] translating storage class for in-tree plugin kubernetes.io/gce-pd to CSI
I0320 18:46:18.982193       1 connection.go:193] GRPC call: /csi.v1.Controller/CreateVolume
I0320 18:46:18.982260       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"task-pv-claim", UID:"b677fa97-f016-4b0f-870b-054f1fb1a6ef", APIVersion:"v1", ResourceVersion:"1828", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "default/task-pv-claim"
I0320 18:46:18.982413       1 connection.go:194] GRPC request: {"accessibility_requirements":{"preferred":[{"segments":{"topology.gke.io/zone":"us-central1-b"}}],"requisite":[{"segments":{"topology.gke.io/zone":"us-central1-b"}}]},"capacity_range":{"required_bytes":5368709120},"name":"pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef","parameters":{"csi.storage.k8s.io/pv/name":"pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef","csi.storage.k8s.io/pvc/name":"task-pv-claim","csi.storage.k8s.io/pvc/namespace":"default","replication-type":"none","type":"pd-standard"},"volume_capabilities":[{"AccessType":{"Mount":{"fs_type":"ext4"}},"access_mode":{"mode":1}}]}
I0320 18:46:20.533232       1 leaderelection.go:278] successfully renewed lease gce-pd-csi-driver/pd-csi-storage-gke-io
I0320 18:46:22.753194       1 connection.go:200] GRPC response: {"volume":{"accessible_topology":[{"segments":{"topology.gke.io/zone":"us-central1-b"}}],"capacity_bytes":5368709120,"volume_id":"projects/songsunny-joonix/zones/us-central1-b/disks/pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef"}}
I0320 18:46:22.753218       1 connection.go:201] GRPC error: <nil>
I0320 18:46:22.753229       1 controller.go:826] create volume rep: {CapacityBytes:5368709120 VolumeId:projects/songsunny-joonix/zones/us-central1-b/disks/pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef VolumeContext:map[] ContentSource:<nil> AccessibleTopology:[segments:<key:"topology.gke.io/zone" value:"us-central1-b" > ] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}
I0320 18:46:22.753379       1 controller.go:898] createVolumeOperation: set annotation [volume.kubernetes.io/provisioner-deletion-secret-namespace/volume.kubernetes.io/provisioner-deletion-secret-name] on pv [pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef].
I0320 18:46:22.753436       1 controller.go:923] successfully created PV pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef for PVC task-pv-claim and csi volume name projects/songsunny-joonix/zones/us-central1-b/disks/pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef
I0320 18:46:22.753507       1 controller.go:939] successfully created PV {GCEPersistentDisk:&GCEPersistentDiskVolumeSource{PDName:pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef,FSType:ext4,Partition:0,ReadOnly:false,} AWSElasticBlockStore:nil HostPath:nil Glusterfs:nil NFS:nil RBD:nil ISCSI:nil Cinder:nil CephFS:nil FC:nil Flocker:nil FlexVolume:nil AzureFile:nil VsphereVolume:nil Quobyte:nil AzureDisk:nil PhotonPersistentDisk:nil PortworxVolume:nil ScaleIO:nil Local:nil StorageOS:nil CSI:nil}
I0320 18:46:22.753648       1 controller.go:1426] provision "default/task-pv-claim" class "in-tree-volume": volume "pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef" provisioned
I0320 18:46:22.753668       1 controller.go:1439] provision "default/task-pv-claim" class "in-tree-volume": succeeded
I0320 18:46:22.753680       1 volume_store.go:154] Saving volume pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef
I0320 18:46:22.761435       1 volume_store.go:157] Volume pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef saved
I0320 18:46:22.761498       1 controller.go:1069] Claim processing succeeded, removing PVC b677fa97-f016-4b0f-870b-054f1fb1a6ef from claims in progress
I0320 18:46:22.761716       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"task-pv-claim", UID:"b677fa97-f016-4b0f-870b-054f1fb1a6ef", APIVersion:"v1", ResourceVersion:"1828", FieldPath:""}): type: 'Normal' reason: 'ProvisioningSucceeded' Successfully provisioned volume pvc-b677fa97-f016-4b0f-870b-054f1fb1a6ef
```